src/dataset/resource.rs

```diff
@@ -15,6 +15,43 @@ impl Resource {
             url,
         }
     }
+
+    pub fn from_type_string(type_name: String, url: String) -> Self {
```

Comment:
I think this would be better served by something like

```rust
impl From<&'_ str> for Type {
  fn from(val: &str) -> Self {
    match val {
      "text/plain" => Self::PlainText,
      ...
      "application/zip" => Self::Archive,
      _ => Self::Unknown,
    }
  }
}
```

which could then be used like

```rust
Resource {
  r#type: document.content_type.into(),
  url: document.url,
}
```

as this more flexible while still providing the same functionality.

Reply:
Yes, that looks much better.